core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -429,8 +459,9 @@ object Cause extends Serializable {
   final def die(defect: Throwable): Cause[Nothing]              = Die(defect)
   final def fail[E](error: E): Cause[E]                         = Fail(error)
   final def interrupt(fiberId: Fiber.Id): Cause[Nothing]        = Interrupt(fiberId)
-  final def stack[E](cause: Cause[E]): Cause[E]                 = Meta(cause, Data(false))
-  final def stackless[E](cause: Cause[E]): Cause[E]             = Meta(cause, Data(true))
+  final def stack[E](cause: Cause[E]): Cause[E]                 = Meta(cause, Data(false, false))
+  final def stackless[E](cause: Cause[E]): Cause[E]             = Meta(cause, Data(true, false))
+  final def silent[E](cause: Cause[E]): Cause[E]                = Meta(cause, Data(false, true))
```

Comment:
We should add `nonSilent` here.

Reply:
agreed.
Also currently silent will also enable stacktraces which I don't believe should be the case?

Reply:
That is a good point. Currently we are forcing every time we call `Meta` to fully specify all the parameters versus just one. I think this would suggest going back to an encoding where `Meta` is a sum type with subtypes for `Stackless` and `Silent` and then we may just have to change the implementation of the pretty print and `stripSilent` functions with regard to how we propagate the information about the current state when we recurse down.